rest_framework_simplejwt/backends.py

```diff
@@ -54,5 +54,7 @@ def decode(self, token, verify=True):
             return jwt.decode(token, self.verifying_key, algorithms=[self.algorithm], verify=verify,
                               audience=self.audience, issuer=self.issuer,
                               options={'verify_aud': self.audience is not None})
+        except InvalidAlgorithmError as ex:
+            raise TokenBackendError(_('Invalid algorithm specified')) from ex
```

Comment:
May I suggest just having us say RS algorithms need cryptography? The PyJWT package seems to check if cryptography is installed in __main__.py which is when InvalidAlgorithmError is raised if the algorithm is not set in DEFAULT_ALGORITHMS for PyJWT.
```suggestion
            raise TokenBackendError(_('Invalid algorithm specified. RS algorithms require cryptography.')) from ex
```
A different, more distinct method would be:
```
            if str(self.algorithm)[:2].lower() == "rs":
                # SimpleJWT supports RS and HS
                raise TokenBackendError(_('You must have cryptography installed to use RS. Other asymmetric encryption algorithms are not supported.'))
            raise TokenBackendError(_('Invalid algorithm specified.')) from ex
```
I wouldn't worry about cleaning or validating self.algorithm since it's up to the developer. The first error is is cryptography is installed. The second error is just an invalid algorithm.

Reply:
> Hey @nicktindall I think the message can be made a little better since this issue sprouted from the cryptography package not being installed. I seem to only find this exception raised here: https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py so it'd be nice to have some additional message regarding the cryptography package.
> 
> I've left my suggestion of what we could try instead.

I had another think about this and given that `jwt.algorithms` exposes `has_crypto` and `requires_cryptography` indicating whether cryptography is installed and which algorithms require it, we can extend the validation of the algorithm performed in the TokenBackend constructor to check firstly that the algorithm is one recognised by simpleJWT and secondly that if it is one that requires cryptography, that cryptography is installed.

I added this validation and used the wording you suggested when the validation fails.

This should mean we never see `InvalidAlgorithmException` raised by `jwt.decode` due to cryptography not being installed. I left the specific handler there for the case where there is a mismatch between the token and the `SIMPLE_JWT` settings.

Reply:
I could see this being an issue if `has_crypto` and `requires_cryptography` were not part of the "public API" of JWT. But the documents appear to be half finished, so it's hard to be sure.

Reply:
I think we'll be fine for now. I doubt they'll remove these attributes in the future. If they do, we can do a backwards compatible move of "copying and pasting" exactly what they did (a try/except, iterable of cryptography necessary stuff, etc.)